File: pid-backend/src/main/java/cz/fit/cvut/pidbackend/Model/Position.java
package cz.fit.cvut.pidbackend.Model;

import javax.persistence.*;
import java.sql.Timestamp;

@Entity
@Table(name = "positions")
public class Position {

    @EmbeddedId
    public PositionId uid;

    @ManyToOne
    @JoinColumn(name = "next_stop_id", referencedColumnName = "uid")
    private Stop nextStopId; // Stop Entity
    @ManyToOne
    @JoinColumn(name = "last_stop_id", referencedColumnName = "uid")
    private Stop prevStopId; // Stop Entity

    @Column(name = "start_timestamp")
    private Timestamp startTimestamp;
    @Column(name = "lat")
    private Double lat;
    @Column(name = "lon")
    private Double lon;
    @Column(name = "trip_sequenceId")
    private Long tripSequenceId;
    @Column(name = "dist_traveled")
    private Double distTraveled;
    @Column(name = "is_canceled")
    private boolean isCanceled;

}

File: android-app/app/src/main/java/cvut/fit/pidmobapp/model/Position.kt
package cvut.fit.pidmobapp.model

import androidx.room.Entity
import androidx.room.PrimaryKey

@Entity
data class Position(
        @PrimaryKey(autoGenerate = true)
        val id: Int,
        val tripId: String, // Trip Entity
        val lat: Double,
        val lon: Double,
        val tripSequenceId: Long,
        val distTraveled: Double,
        val nextStopId: String, // Stop Entity
        val prevStopId: String, // Stop Entity
        val isCanceled: Boolean
)
